src/web/mjs/engine/Connector.mjs

```diff
@@ -64,7 +65,11 @@ export default class Connector {
     async initialize() {
         try {
             if(!this.initialized) {
-                await this._initializeConnector();
+                if(!this.initializing)
```

Comment:
Avoid nested conditions if possible.
When initialization fails, the flag is not reset and therefore the connector can never be initialized.

suggestion, something like this
```javascript
if(!this.initialized && !this.initializing) {
    this.initializing = true;
    await this._initializeConnector();
    this.initialized = true;
}
// use a finally for the try block
} finally {
    this.initializing = false;
}
```